controllers/complianceeventsapi/server.go

```diff
@@ -802,6 +795,204 @@ func postComplianceEvent(db *sql.DB, w http.ResponseWriter, r *http.Request) {
 	}
 }
 
+func getComplianceEventsQuery(whereClause string, queryArgs *queryOptions) string {
+	// Example query:
+	//   SELECT compliance_events.id, compliance_events.compliance, ...
+	//     FROM compliance_events
+	//   LEFT JOIN clusters ON compliance_events.cluster_id = clusters.id
+	//   LEFT JOIN parent_policies ON compliance_events.parent_policy_id = parent_policies.id
+	//   LEFT JOIN policies ON compliance_events.policy_id = policies.id
+	//   WHERE (policies.name=$1 OR policies.name=$2) AND (policies.kind=$3)
+	//   ORDER BY compliance_events.timestamp desc
+	//   LIMIT 20
+	//   OFFSET 0 ROWS;
+	return fmt.Sprintf(`%s%s
+	ORDER BY %s %s
+	LIMIT %d
+	OFFSET %d ROWS;`,
+		generateGetComplianceEventsQuery(queryArgs.IncludeSpec),
+		whereClause,
+		strings.Join(queryArgs.Sort, ", "),
+		queryArgs.Direction,
+		queryArgs.PerPage,
+		(queryArgs.Page-1)*queryArgs.PerPage,
+	)
+}
+
+func getComplianceEventsCSV(db *sql.DB, w http.ResponseWriter, r *http.Request) {
+	queryArgs, err := parseQueryArgs(r.URL.Query())
+	if err != nil {
+		writeErrMsgJSON(w, err.Error(), http.StatusBadRequest)
+
+		return
+	}
+
+	// Note that the where clause could be an empty string if no filters were passed in the query arguments.
+	whereClause, filterValues := getWhereClause(queryArgs)
+
+	query := getComplianceEventsQuery(whereClause, queryArgs)
+
+	rows, err := db.QueryContext(r.Context(), query, filterValues...)
+	if err == nil {
+		err = rows.Err()
+	}
+
+	if err != nil {
+		log.Error(err, "Failed to query for compliance events")
+		writeErrMsgJSON(w, "Internal Error", http.StatusInternalServerError)
+
+		return
+	}
+
+	defer rows.Close()
+
+	titles := []string{
+		"compliance_events_id",
+		"compliance_events_compliance",
+		"compliance_events_message",
+		"compliance_events_metadata",
+		"compliance_events_reported_by",
+		"compliance_events_timestamp",
+		"clusters_cluster_id",
+		"clusters_name",
+		"parent_policies_id",
+		"parent_policies_name",
+		"parent_policies_namespace",
+		"parent_policies_categories",
+		"parent_policies_controls",
+		"parent_policies_standards",
+		"policies_id",
+		"policies_api_group",
+		"policies_kind",
+		"policies_name",
+		"policies_namespace",
+		"policies_severity",
+		"policies_spec",
```

Comment:
Does `policies_spec` need to be conditional based on if `queryArgs.IncludeSpec` is set or does the library handle this automatically by dropping this header?

Reply:
it is going to be nil

Reply:
It is going to be "nil"

Reply:
Do you want drop the header?

Reply:
Yeah, let's drop the header if `queryArgs.IncludeSpec` is not true.